From: YOKOTA Hiroshi <yokota.hgml@gmail.com>
Date: Fri, 17 Nov 2023 15:36:02 +0900
Subject: Revert "When saving PDF in PoDoFo dont collect garbage to avoid
 slowdowns with some PDF files"

Forwarded: not-needed

This reverts commit e8d3e372087e87139360104bff250747ae3a3367.
---
 src/calibre/utils/podofo/doc.cpp  | 4 ++++
 src/calibre/utils/podofo/global.h | 7 ++++---
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/calibre/utils/podofo/doc.cpp b/src/calibre/utils/podofo/doc.cpp
index 648a90c..b174ea6 100644
--- a/src/calibre/utils/podofo/doc.cpp
+++ b/src/calibre/utils/podofo/doc.cpp
@@ -6,8 +6,12 @@
  */
 
 #include "global.h"
+#include <iostream>
 #include <algorithm>
 #include <new>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
 
 using namespace pdf;
 
diff --git a/src/calibre/utils/podofo/global.h b/src/calibre/utils/podofo/global.h
index 1fcb0f0..b1b1e69 100644
--- a/src/calibre/utils/podofo/global.h
+++ b/src/calibre/utils/podofo/global.h
@@ -13,6 +13,7 @@
 #define USING_SHARED_PODOFO
 #include <podofo.h>
 #include <unordered_set>
+#include <unordered_map>
 using namespace PoDoFo;
 using namespace std::literals;
 
@@ -129,9 +130,9 @@ object_as_reference(const PdfObject *o) {
     return o->IsReference() ? o->GetReference() : o->GetIndirectReference();
 }
 
-// NoMetadataUpdate needed to avoid PoDoFo clobbering the /Info and XMP metadata with its own nonsense
-// NoCollectGarbage needed to avoid slowdowns on writing, see https://github.com/podofo/podofo/issues/99#issuecomment-1787239479
-static const PdfSaveOptions save_options = PdfSaveOptions::NoMetadataUpdate | PdfSaveOptions::NoCollectGarbage;
+// Needed to avoid PoDoFo clobbering the /Info and XMP metadata with its own nonsense
+// rename to NoMetadataUdate after https://github.com/podofo/podofo/commit/96689eb6e45b71eae1577ecb2d4a796c52e9a813
+static const PdfSaveOptions save_options = PdfSaveOptions::NoModifyDateUpdate;
 
 class PdfReferenceHasher {
     public:
